src/evict/evict_lru.c

```diff
@@ -907,24 +907,7 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
 	/* Get some more pages to consider for eviction. */
 	if ((ret = __evict_walk(cache->walk_session, queue)) == EBUSY)
 		return (0);	/* An interrupt was requested, give up. */
-	WT_RET_NOTFOUND_OK(ret);
-
-	/*
-	 * If we found no pages at all during the walk, something is wrong.
-	 * Be more aggressive next time.
-	 *
-	 * Continue on to sort the queue, in case there are pages left from a
-	 * previous walk.
-	 */
-	if (ret == WT_NOTFOUND) {
-		if (F_ISSET(cache,
-		    WT_CACHE_EVICT_CLEAN_HARD | WT_CACHE_EVICT_DIRTY_HARD))
-			cache->evict_aggressive_score = WT_MIN(
-			    cache->evict_aggressive_score + WT_EVICT_SCORE_BUMP,
-			    WT_EVICT_SCORE_MAX);
-		WT_STAT_CONN_SET(session, cache_eviction_aggressive_set,
-		    cache->evict_aggressive_score);
-	}
```

Comment:
@michaelcahill I didn't end up merging this branch today (and therefore didn't update MongoDB master). The reason I didn't is because I couldn't get the performance of update-btree back without removing this change - i.e: the change to no longer return WT_NOTFOUND from ``__evict_walk``. I tested the various cases I have around related to WT-2954, and none of them relied on this part of the pull request - I'm not sure what case it improved.

I also couldn't get to the bottom of why adding it back in helps - generally switching into aggressive mode hurts the update-btree workload, and this change makes that happen less often.